packages/@lwc/template-compiler/src/codegen/codegen.ts

```diff
@@ -96,6 +98,17 @@ export default class CodeGen {
         }
     }
 
+    genComment(parts: Array<string | t.Expression>, dynamicIndexes: number[]): t.Expression {
+        const transformedParts = parts.map((part) =>
+            typeof part === 'string' ? t.literal(part) : part
+        );
+
+        return this._renderApiCall(RENDER_APIS.comment, [
+            t.arrayExpression(transformedParts),
+            t.arrayExpression(dynamicIndexes.map((idx) => t.literal(idx))),
```

Comment:
Why is the second argument needed?

Reply:
These are the positions that are dynamic (`{}`); they are used to escape the strings in ssr. (on dom context, the DOM API takes carer of that)